Analysis/src/QwPromptSummary.cc

```diff
@@ -368,45 +327,40 @@ QwPromptSummary::LoadElementsFromParameterFile(QwParameterFile& parameterfile)
     delete section;
   }
   
-  QwMessage << "QwPromptSummary: Loaded " << fNElements << " elements from parameter file" << QwLog::endl;
+  QwMessage << "QwPromptSummary: Loaded " << fElementList.size() << " elements from parameter file" << QwLog::endl;
 };
 
 
 void 
 QwPromptSummary::AddElement(PromptSummaryElement *in)
 {
-  Int_t pos = 0;
-
-  fElementList.push_back(in);
+  TString name = in->GetName(); name.ToLower();
+  fElementList[name] = in;
   if(fLocalDebug) {
-    printf("AddElement at pos %d\n", pos);
+    printf("AddElement %s at pos %lu\n", in->GetName().Data(), fElementList.size()-1);
+  }
+  if (fElementList.size() == 1) {
+    QwMessage << "First element: " << in->GetName() << " will be reference" << QwLog::endl;
+    fReferenceElement = in;
   }
-
-  fNElements++;
-  return;
 };
 
 
 PromptSummaryElement *
 QwPromptSummary::GetElementByName(TString name)
 {
-
-  
-  TString get_name = "";
-
-  for (auto i=fElementList.begin(); i!=fElementList.end(); i++  )
-    {
-      PromptSummaryElement* an_element = *i;
-      get_name   = an_element->GetName();
-      if( get_name.EqualTo(name) ) {
-	if(fLocalDebug) {
-	  std::cout << "System " << an_element->GetName()
-		    << " QwPromptSummary::GetElementByName address at" << an_element << std::endl;
-	}
-	return an_element;
-      }
+  name.ToLower();
+  if (fElementList.find(name) != fElementList.end()) {
+    PromptSummaryElement* an_element = fElementList[name];
+    if(fLocalDebug) {
+      std::cout << "System " << an_element->GetName()
+                << " QwPromptSummary::GetElementByName address at " << an_element << std::endl;
     }
-
+    return an_element;
+  } else {
+      std::cout << "System " << name
+                << " QwPromptSummary::GetElementByName not found" << std::endl;
```

Comment:
This should probably be a QwDebug output instead of cout.  All elements in all subsystems and data handlers get run through this function, producing a lot of output lines that are not particularly meaningful.
Or maybe wrap it with the "fLocalDebug" like in lines 356-357.

Reply:
```suggestion
      QwDebug << "System " << name
              << " QwPromptSummary::GetElementByName not found" << std::endl;
```